corporate/lib/stripe.py

```diff
@@ -252,14 +262,18 @@ def next_invoice_date(plan: CustomerPlan) -> Optional[datetime]:
     return dt
 
 
-def renewal_amount(plan: CustomerPlan, event_time: datetime) -> int:  # nocoverage: TODO
+def renewal_amount(
+    plan: CustomerPlan, event_time: datetime, last_ledger_entry: Optional[LicenseLedger] = None
+) -> int:  # nocoverage: TODO
     if plan.fixed_price is not None:
         return plan.fixed_price
-    realm = plan.customer.realm
-    billing_session = RealmBillingSession(user=None, realm=realm)
-    new_plan, last_ledger_entry = billing_session.make_end_of_cycle_updates_if_needed(
-        plan, event_time
-    )
+    new_plan = None
+    if last_ledger_entry is None:
+        realm = plan.customer.realm
+        billing_session = RealmBillingSession(user=None, realm=realm)
+        new_plan, last_ledger_entry = billing_session.make_end_of_cycle_updates_if_needed(
+            plan, event_time
+        )
```

Comment:
I have added `last_ledger_entry` optional argument temporarily to avoid calling `make_end_of_cycle_updates_if_needed` twice in `billing_home`. The long term fix related to `estimate_annual_recurring_revenue_by_realm`, we discussed in CZO would be a follow-up.

This unblocks the work in this direction.

Reply:
This works for me; and I can imagine even `estimate_annual_recurring_revenue_by_realm` working by passing the `last_ledger_entry` parameter here as how we land there.